cdap-api-common/src/main/java/co/cask/cdap/api/data/format/RecordFormat.java

```diff
@@ -77,7 +77,11 @@ public void initialize(FormatSpecification formatSpecification) throws Unsupport
     }
     desiredSchema = desiredSchema == null ? getDefaultSchema() : desiredSchema;
     if (desiredSchema == null) {
-      throw new UnsupportedTypeException("A schema must be provided to this format.");
+      String msg = "A schema must be provided to the format: ";
+      if (formatSpecification != null) {
```

Comment:
its not marked ``@Nullable``, but it has a null check right above this code block, so Intellij warns.